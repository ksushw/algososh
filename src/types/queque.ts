export interface IQueue<T> {
  enqueue: (item: T) => void;
  dequeue: () => void;
  peak: () => void;
  getSize: () => number;
  getStack: () => Array<T | "">;
}
