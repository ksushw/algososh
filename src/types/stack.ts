export interface IStack<T> {
  push: (item: T) => void;
  pop: () => void;
  peak: () => void;
  getSize: () => number;
  getStack: () => Array<T>;
}
