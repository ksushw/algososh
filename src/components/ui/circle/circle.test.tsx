import renderer from "react-test-renderer";
import { Circle } from "./circle";
import { ElementStates } from "../../../types/element-states";

it("Кружок без буквы рендериться без ошибок", () => {
  const tree = renderer.create(<Circle />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("Кружок с буквами рендериться без ошибок", () => {
  const tree = renderer.create(<Circle letter={"exmp"} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("Кружок с head  рендериться без ошибок", () => {
  const tree = renderer.create(<Circle head={"head"} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("Кружок с react-элементом в head  рендериться без ошибок", () => {
  const tree = renderer
    .create(<Circle head={<Circle letter="exmp" />} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("Кружок с tail рендериться без ошибок", () => {
  const tree = renderer.create(<Circle tail={"tail"} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("Кружок с react-элементом в tail рендериться без ошибок", () => {
  const tree = renderer
    .create(<Circle tail={<Circle letter="exmp" />} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("Кружок с index рендериться без ошибок", () => {
  const tree = renderer
    .create(<Circle tail={<Circle index={1234} />} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("Кружок с пропом isSmall ===  true рендериться без ошибок", () => {
  const tree = renderer
    .create(<Circle tail={<Circle isSmall={true} />} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("Кружок в состоянии default рендериться без ошибок", () => {
  const tree = renderer
    .create(<Circle tail={<Circle state={ElementStates.Default} />} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("Кружок в состоянии changing рендериться без ошибок", () => {
  const tree = renderer
    .create(<Circle tail={<Circle state={ElementStates.Changing} />} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("Кружок в состоянии modified рендериться без ошибок", () => {
  const tree = renderer
    .create(<Circle tail={<Circle state={ElementStates.Modified} />} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
