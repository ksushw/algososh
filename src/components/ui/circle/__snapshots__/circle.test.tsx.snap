// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Кружок без буквы рендериться без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Кружок в состоянии changing рендериться без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   changing"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
</div>
`;

exports[`Кружок в состоянии default рендериться без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
</div>
`;

exports[`Кружок в состоянии modified рендериться без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   modified"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
</div>
`;

exports[`Кружок с head  рендериться без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    head
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Кружок с index рендериться без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      >
        1234
      </p>
      <div
        className="text text_type_input text_color_input mt-4 absolute tail60 element"
      />
    </div>
  </div>
</div>
`;

exports[`Кружок с react-элементом в head  рендериться без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        >
          exmp
        </p>
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Кружок с react-элементом в tail рендериться без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        >
          exmp
        </p>
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
</div>
`;

exports[`Кружок с tail рендериться без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    tail
  </div>
</div>
`;

exports[`Кружок с буквами рендериться без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      exmp
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Кружок с пропом isSmall ===  true рендериться без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle  small default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
</div>
`;
